pkgs/top-level/perl-packages.nix

```diff
@@ -28609,6 +28594,35 @@ with self; {
     };
   };
 
+  Xapian = buildPerlModule rec {
+    pname = "Xapian";
+    version = "1.4.27";
+    src = fetchurl {
+      url = "https://oligarchy.co.uk/xapian/${version}/xapian-bindings-${version}.tar.xz";
+      sha256 = "1fhq6rydjymmyn79cdza0j4rmlizrrwmf5mx276rlmwyh085wfxs";
+    };
+    buildInputs = [ pkgs.xapian ];
+    preConfigure = ''
+      # FIXME: doesn't work for cross
+      export PERL_LIB="$out/lib/perl5/site_perl/${perl.version}"
+      export PERL_ARCH="$PERL_LIB/$(perl -MConfig -e 'print $Config{archname}')"
+    '';
+    configureFlags = [
+      "--with-perl"
+    ];
+    outputs = [ "out" ]; # no "devdoc"
+    # Use default phases
```

Comment:
Would it make more sense to use buildPerlPackage then?

Reply:
In theory yes, but I think you can't disable its `preConfigure` hook that tries to execute `Makefile.pl`.

Reply:
@sternenseemann I don't understand everything, but maybe we could create a dummy `Makefile.PL` if the problem is that it's missing, I see that `buildPerlModule` does something like that:

https://github.com/NixOS/nixpkgs/blob/62b32a63f5027b6bbc3db0d15252434eeebff489/pkgs/top-level/perl-packages.nix#L66-L69